woocommerce/woocommerce-android
# Feature/9332 blaze banner visibility logic

Part of: #9332 

### Description

This PR is a mix of small changes needed to wrap up the new Blaze banner feature.

**Show/hide the banner according to the following criteria**

- The site must have at least one product published
- The site must be public/launched
- If user creates a Blaze campaign then both banners should be hidden.
- The user has not dismissed any of the banners previously. If one of the banners is dismissed by clicking on the "X" button then both banners will be hidden.

**Fix UI glitches**

- Sets the correct alignment between the "X" button and the flame icon
- Sets the correct size and color to the "X" (24dp gray)
- Sets bold text for the main CTA

**Bugfix**: Avoid displaying Blaze option in the overflow menu options inside product detail when the product is draft

### Testing instructions

**Test bugfix**
1. Log into a site that is Blaze eligible (user `isAdmin`, site is atomic and Jetpack connected)  
2. Click on add new product
3. Click on the overflow menu of the product creation screen and check the Blaze option is gone
4. Publish the product
5. Check Blaze option is visible

**Test Blaze banner visibility checks**
1. Log into a site that is Blaze eligible (user `isAdmin`, site is atomic and Jetpack connected) 
2. Check the Blaze banner is displayed in Product list and My Store tab only if the following is met: 
```
- The site must have at least one product published
- The site must be public/launched
- If user creates a Blaze campaign then both banners should be hidden.
- The user has not dismissed any of the banners previously. If one of the banners is dismissed by clicking on the "X" button then both banners will be hidden.
```
3. Dismiss banner and check the following event is tracked: `Tracked: blaze_banner_dismissed, Properties: {"source":"my_store_banner","blog_id":205513046,"is_wpcom_store":false,"is_debug":true}`
4. Check the banner is never shown again unless you log out and log in again.
